src/index.js

```diff
@@ -60,20 +60,54 @@ export default function boolean(subject, clipping, operation) {
   const sbbox = [Infinity, Infinity, -Infinity, -Infinity];
   const cbbox = [Infinity, Infinity, -Infinity, -Infinity];
 
-  //console.time('fill queue');
+  // console.time('fill queue');
   const eventQueue = fillQueue(subject, clipping, sbbox, cbbox, operation);
   //console.timeEnd('fill queue');
 
   trivial = compareBBoxes(subject, clipping, sbbox, cbbox, operation);
   if (trivial) {
     return trivial === EMPTY ? null : trivial;
   }
-  //console.time('subdivide edges');
+  // console.time('subdivide edges');
   const sortedEvents = subdivideSegments(eventQueue, subject, clipping, sbbox, cbbox, operation);
   //console.timeEnd('subdivide edges');
 
-  //console.time('connect vertices');
+  // console.time('connect vertices');
   const result = connectEdges(sortedEvents, operation);
   //console.timeEnd('connect vertices');
-  return result;
+
+  const out = [];
+  for (var i = 0; i < result.length; i++) {
+    const contour = result[i];
+    const points = contour.points;
+    if (contour.external) {
+      if (points[0][0] !== points[points.length - 1][0] ||
+          points[0][1] !== points[points.length - 1][1]
```

Comment:
Why is this check needed (and the one in line 96)? Should the rings produced by the algorithm not be closed in any case? Or is this merely needed for the currently broken test cases that sometimes don't close the input polygons properly? If that's the case we should rather take care of that upfront.